### DISCUSSION TITLE
JSON Schema 规范对齐的问题

### DISCUSSION THREAD

**Discussion by zhangjv** [alibaba/formily#2352] [Q&A 2.x] [answered]
>  仅涉及Schema Form场景

formily 遵循 JSON Schema 规范作为自身的配置规范，并且实现大部分规范所描述校验逻辑，见  [<SchemaMarkupField/> props](https://formilyjs.org/#/zoi8i0/6dt3t7FjI4)

受益于遵循标准规范，服务端依据同一套JSON Schema配置校验前端提交过来的表单数据是否合法成为可能，且目前业界已有不少成熟的校验库可以直接复用

问题在于： JSON Schema 规范自身也在发展，formily 采用的某些属性在最新的规范中已经被移除，参见 [https://json-schema.org/draft/2020-12/json-schema-validation.html ](https://json-schema.org/draft/2020-12/json-schema-validation.html ) 11.2. Informative References
移除的字段包括："allOf", "anyOf", "oneOf", "not", "if", "then", "else", "items", "additionalItems", "contains", "propertyNames", "properties", "patternProperties", "additionalProperties" 等

其中，比如formily的array类型用items属性存储array内部字段，属于上一版规范中的定义，在最新版已经移除。也就是说如果服务端也**基于formily的这套表单配置、用最新规范校标准验用户提交的表单内容**，就不能有效校验array的内部字段

所以未来是要对齐规范，还是保持目前的配置结构稳定不变？

**Comment by janryWang** [comment, marked as answer]
不可能完全follow标准的，这套schema是formily json-schema
